extruct/uniform.py

```diff
@@ -1,16 +1,28 @@
 from six.moves.urllib.parse import urlparse, urljoin
 
 
-def _uopengraph(extracted):
+def _uopengraph(extracted, with_og_arr=False):
     out = []
     for obj in extracted:
-        # In order of appearance in the page
         properties = list(reversed(obj['properties']))
-        # Ensuring that never empty value is returned if there is a duplicated
-        # property with non empty value
+        # Set of non empty properties
         non_empty_props = {k for k, v in properties if v and v.strip()}
+        # Set of repeated properties with at least 2 non empty values
+        repeated_props = {}
+        if with_og_arr:
+            repeated_props = {k for k in non_empty_props
+                              if len([i for i, v in properties if i == k and (v and v.strip())]) > 1}
+        # Add properties that either have only empty values or are duplicated and
+        # have only 1 non empty value
         flattened = {k: v for k, v in properties
-                     if k not in non_empty_props or (v and v.strip())}
+                     if k not in repeated_props and (k not in non_empty_props or (v and v.strip()))}
```

Comment:
I think we should not exclude values if they are in repeated properties, at least when the ``with_og_arr`` key is not set. Would you also mind adding a test which checks this?

Reply:
Actually on line 11 `repeated_props` is set to be empty so if `with_og_arr` is not set this will work fine. I could add a test case to check this but I think [one](https://github.com/scrapinghub/extruct/blob/master/tests/test_uniform.py#L12) exists already